api/serializers.py

```diff
@@ -72,6 +72,33 @@ class Meta:
     legacy = LegacySMSMediaSerializer(
         source='sms', help_text='Information from legacy SMS', required=False, read_only=True)
 
+    def create(self, validated_data):
+        """
+        Override behaviour when creating a new object using this serializer. If the current request
+        is being passed in the context, give the request user edit and view permissions on the
+        item.
+
+        """
+        new_item = super().create(validated_data)
+
+        if self.context is not None and 'request' in self.context:
+            request = self.context['request']
+            if not request.user.is_anonymous:
+                # Due to Django ORM oddness, we need to re-fetch the object to correctly modify
+                # permissions otherwise the ORM gets confused
+                new_item = (
+                    mpmodels.MediaItem.objects.all()
+                    .only()
+                    .select_related('view_permission', 'edit_permission')
+                    .get(id=new_item.id)
+                )
+                new_item.view_permission.crsids.append(request.user.username)
```

Comment:
doesn't the post_save does this at this moment? Or is this to have more completeness?

Reply:
The post_save hook merely makes sure that an associated permission object is created in the database. This makes sure it has the current user set on the permission. (Note that no permission object is created here.) Since this may be a common pattern, I'll hoist it into an object manager method.